File: samples/JavaScript/towers-of-hanoi.js
function move(n, a, b, c) {
	if (n > 0) {
		move(n-1, a, c, b);
		console.log("Move disk from " + a + " to " + c);
		move(n-1, b, a, c);
	}
}
move(4, "A", "B", "C");

File: samples/Rust/towers-of-hanoi.rs
fn move(n: int, from: int, to: int, via: int) {
  if (n > 0) {
    move(n - 1, from, via, to);
    println!("Move disk from pole {:d} to pole {:d}", from, to);
    move(n - 1, via, to, from);
  }
}

fn main() {
  move(4, 1,2,3);
}
